src/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -6372,6 +6379,320 @@ HRESULT ProfToEEInterfaceImpl::GetFunctionInfo2(FunctionID funcId,
+    }
+
+    return S_OK;
+}
+
+/*
+* GetNoMetadataFunctionInfo
+*
+* This function takes a functionId that maybe of a metadata-less method like an IL Stub
+* or LCG method and gives information about it without failing like GetFunctionInfo.
+*
+* Parameters:
+*   functionId - The function that is being requested.
+*   pModuleId - An optional parameter for returning the module of the function.
+*   ppMethodHeader - An optional parameter for returning the IL header of the function.
+*   pcbMethodSize - An optional parameter for returning the size of the IL header of the function.
+*   ppvSig -  An optional parameter for returning the signature of the function.
+*   pbSig - An optional parameter for returning the size of the signature of the function.
+*   cchName - A parameter for indicating the size of buffer for the wszName parameter.
+*   pcchName - An optional parameter for returning the true size of the wszName parameter.
+*   wszName - A parameter to the caller allocated buffer of size cchName
+*
+* Returns:
+*   S_OK if successful.
+*/
+HRESULT ProfToEEInterfaceImpl::GetNoMetadataFunctionInfo(FunctionID functionId,
+    ModuleID *pModuleId,
+    LPCBYTE* ppMethodHeader,
+    ULONG* pcbMethodSize,
+    PCCOR_SIGNATURE* ppvSig,
+    ULONG* pbSig,
+    ULONG cchName,
+    ULONG *pcchName,
+    __out_ecount_part_opt(cchName, *pcchName) WCHAR wszName[])
+{
+    CONTRACTL
+    {
+    NOTHROW;
+    GC_NOTRIGGER;
+    MODE_ANY;
+    EE_THREAD_NOT_REQUIRED;
+
+    // Generics::GetExactInstantiationsOfMethodAndItsClassFromCallInformation eventually
+    // reads metadata which causes us to take a reader lock.  However, see
+    // code:#DisableLockOnAsyncCalls
+    DISABLED(CAN_TAKE_LOCK);
+
+    // Asynchronous functions can be called at arbitrary times when runtime 
+    // is holding locks that cannot be reentered without causing deadlock.
+    // This contract detects any attempts to reenter locks held at the time 
+    // this function was called.  
+    CANNOT_RETAKE_LOCK;
+
+    SO_NOT_MAINLINE;
+
+    PRECONDITION(CheckPointer(pModuleId, NULL_OK));
+    PRECONDITION(CheckPointer(ppvSig, NULL_OK));
+    PRECONDITION(CheckPointer(pbSig,  NULL_OK));
+    PRECONDITION(CheckPointer(pcchName, NULL_OK));
+    }
+    CONTRACTL_END;
+
+    // See code:#DisableLockOnAsyncCalls
+    PERMANENT_CONTRACT_VIOLATION(TakesLockViolation, ReasonProfilerAsyncCannotRetakeLock);
+
+    PROFILER_TO_CLR_ENTRYPOINT_ASYNC_EX(kP2EEAllowableAfterAttach,
+        (LF_CORPROF,
+            LL_INFO1000,
+            "**PROF: GetNoMetadataFunctionInfo 0x%p.\n",
+            functionId));
+
+    //
+    // Verify parameters.
+    //
+
+    if (functionId == NULL)
+    {
+        return E_INVALIDARG;
+    }
+
+    MethodDesc *pMethDesc = FunctionIdToMethodDesc(functionId);
+
+    if (pMethDesc == NULL)
+    {
+        return E_INVALIDARG;
+    }
+
+    // it's not safe to examine a methoddesc that has not been restored so do not do so
```

Comment:
You mean before checking for resotration? This function is to determine whether it is a metadata function or not.